### DISCUSSION TITLE
Attempt to read property "photo_id" on null

### DISCUSSION THREAD

**Discussion by grisu48** [LycheeOrg/Lychee#2394] [Bugs]
### Lychee version

v5.2.1

### Which PHP version are you using?

PHP 8.3

### Detailed description of the problem

After updating to the docker container v5.1.2, my lychee instance on https://photo.feldspaten.org/ throws a 500 error on some galleries with the error message

"Attempt to read property "photo_id" on null"

See e.g. https://photo.feldspaten.org/gallery/2_H3gJ4I9pS3HZUpmnucS2aj

I run lychee with the docker container as a `podman` systemd service. I changed the version tag from 5.1.0 to 5.2.1 and after the update this issue arose.

### Steps to reproduce the issue

Couldn't identify a clean reproducer, but see e.g. https://photo.feldspaten.org/gallery/2_H3gJ4I9pS3HZUpmnucS2aj

### Diagnostics [REQUIRED]

    Diagnostics
    -----------
        Warning: Dropbox import not working. dropbox_key is empty.
        Info: Latest version of PHP is 8.3
        Warning: Default timezone not properly set; you might experience strange results when importing photos without explicit EXIF timezone
        Warning: git (software) is not available.
        Info: Found 5072 small images without filesizes.
              You can use `php artisan lychee:variant_filesize 5072` to compute them.
    

    System Information
    ------------------
        Lychee Version (tag):                    v5.2.1 (acecdcf) -- Up to date (0 seconds ago).
        DB Version:                              5.2.1
        
        composer install:                        --no-dev
        APP_ENV:                                 production
        APP_DEBUG:                               false
        APP_URL:                                 set
        APP_DIR:                                 default
        LOG_VIEWER_ENABLED:                      true
        LIVEWIRE_ENABLED:                        true
        
        System:                                  Linux
        PHP Version:                             8.2.18
        PHP User agent:                          Lychee/4 (https://lycheeorg.github.io/)
        Timezone:                                UTC
        Max uploaded file size:                  100M
        Max post size:                           100M
        Livewire chunk size:                     12.00 MB
        Max execution time:                      3600
        SQLite Version:                          3.40.1
        
        exec() Available:                        yes
        Imagick Available:                       1
        Imagick Enabled:                         1
        Imagick Version:                         1691
        GD Version:                              2.3.3
        Number of foreign key:                   10 found.
    

    Config Information
    ------------------
        version:                                 050201
        check_for_updates:                       0
        sorting_photos_col:                      taken_at
        sorting_photos_order:                    ASC
        sorting_albums_col:                      max_taken_at
        sorting_albums_order:                    ASC
        imagick:                                 1
        skip_duplicates:                         0
        small_max_width:                         0
        small_max_height:                        360
        medium_max_width:                        1920
        medium_max_height:                       1080
        lang:                                    en
        image_overlay_type:                      desc
        default_license:                         CC-BY-NC-ND-4.0
        compression_quality:                     90
        grants_full_photo_access:                1
        delete_imported:                         0
        mod_frame_enabled:                       1
        mod_frame_refresh:                       30
        thumb_2x:                                1
        small_2x:                                1
        medium_2x:                               1
        landing_page_enable:                     0
        force_migration_in_production:           0
        search_public:                           0
        hide_version_number:                     0
        SL_enable:                               0
        SL_for_admin:                            0
        recent_age:                              1
        grants_download:                         1
        photos_wraparound:                       1
        map_display:                             0
        zip64:                                   1
        map_display_public:                      0
        map_provider:                            Wikimedia
        force_32bit_ids:                         0
        map_include_subalbums:                   0
        update_check_every_days:                 3
        has_exiftool:                            1
        share_button_visible:                    0
        import_via_symlink:                      0
        has_ffmpeg:                              1
        location_decoding:                       0
        location_decoding_timeout:               30
        location_show:                           1
        location_show_public:                    0
        rss_enable:                              0
        rss_recent_days:                         7
        rss_max_items:                           100
        prefer_available_xmp_metadata:           0
        editor_enabled:                          1
        lossless_optimization:                   0
        swipe_tolerance_x:                       150
        swipe_tolerance_y:                       250
        local_takestamp_video_formats:           .avi|.mov
        log_max_num_line:                        1000
        unlock_password_photos_with_url_param:   0
        nsfw_visible:                            1
        nsfw_blur:                               0
        nsfw_warning:                            0
        nsfw_warning_admin:                      0
        nsfw_banner_override:                    
        map_display_direction:                   1
        album_subtitle_type:                     oldstyle
        upload_processing_limit:                 4
        new_photos_notification:                 0
        legacy_id_redirection:                   1
        zip_deflate_level:                       6
        SA_enabled:                              1
        default_album_protection:                1
        allow_username_change:                   1
        album_decoration:                        layers
        album_decoration_orientation:            row
        auto_fix_orientation:                    1
        use_job_queues:                          0
        random_album_id:                         starred
        use_last_modified_date_when_no_exif_date: 0
        layout:                                  justified
        date_format_photo_thumb:                 M j, Y, g:i:s A e
        date_format_photo_overlay:               M j, Y, g:i:s A e
        date_format_sidebar_uploaded:            M j, Y, g:i:s A e
        date_format_sidebar_taken_at:            M j, Y, g:i:s A e
        date_format_hero_min_max:                F Y
        date_format_hero_created_at:             M j, Y, g:i:s A T
        date_format_album_thumb:                 M Y
        upload_chunk_size:                       0
        nsfw_banner_blur_backdrop:               0
        search_pagination_limit:                 1000
        search_minimum_length_required:          4
        photo_layout_justified_row_height:       320
        photo_layout_masonry_column_width:       300
        photo_layout_grid_column_width:          250
        photo_layout_square_column_width:        200
        photo_layout_gap:                        12
        display_thumb_album_overlay:             always
        display_thumb_photo_overlay:             hover
        default_album_thumb_aspect_ratio:        1/1
        use_album_compact_header:                0
        login_button_position:                   left
        back_button_enabled:                     0
        back_button_text:                        Return to Home
        current_job_processing_visible:          1

### Browser & System [REQUIRED]

openSUSE Tumbleweed, Firefox 124.0.2

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by ildyria** [comment]
I'll try to have a look tomorrow.

**Reply by grisu48** [reply]
Thank you!

**Reply by ildyria** [reply]
You can thank @nanawel instead. They are the one who figured it out first.

TL;DR: there are no landscape pictures in your album, hence the crash.
I will try to improve the code so that in such events, we still have a header picture (but it will be a crop of a portrait one)

**Comment by ildyria** [comment]
Fixed in 5.2.2

**Comment by grisu48** [comment]
Amazing, thank you!

**Reply by grisu48** [reply]
Confirmed fixed :rocket:
